### PR TITLE
SAAS-914 Ticket Overview

### PR DESCRIPTION
**Pull request by TomasBlazek3PG**
https://files.slack.com/files-pri/T03FRR0BJ-F03GAFERYNL/image.png

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
   "publishConfig": {
     "access": "public"
   },
+  "proxy": "https://portal-dev.percona.com",
```
Comment (fabio-silva) [inline package.json:17]: I believe this is unintended

File: src/components/Charts/Pie/Pie.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        startAngle: p.startAngle,
+        endAngle: p.endAngle,
+      }),
+    );
+  }, [pie, size, width]);
+
+  return (
+    <div className={styles.wrapper}>
+      <svg width={size} height={size}>
+        {/* eslint-disable-next-line no-magic-numbers */}
```
Comment (fabio-silva) [inline src/components/Charts/Pie/Pie.tsx:36]: Let's disable the rule for the whole file instead

File: src/components/SupportTicketOverview/Legend/Legend.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+export const Legend: FC<ILegend> = ({ values }) => {
+  const styles = useStyles(getStyles);
+
+  return (
+    <div className={styles.legend}>
+      <div className={styles.row}>
+        {values.map((item) => (
+          <div className={styles.item}>
+            <div className={styles.colorNameNumber} key={item.department}>
```
Comment (fabio-silva) [inline src/components/SupportTicketOverview/Legend/Legend.tsx:18]: Should key be here or on the parent instead?

File: src/components/SupportTicketOverview/Legend/Legend.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+              <span className={styles.color} style={{ backgroundColor: item.color }} />
+              {item.department}
+              <strong>{item.ammount}</strong>
+            </div>
+          </div>
+        ))}
+      </div>
+      <div className={styles.row}>
+        {values.map((item) => (
+          <div className={styles.item}>
```
Comment (fabio-silva) [inline src/components/SupportTicketOverview/Legend/Legend.tsx:28]: missing key

File: src/components/SupportTicketOverview/SupportTicketOverview.tsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+    () =>
+      computedItems.map((item) => ({
+        name: item.department,
+        value: item.ammount,
+        color: item.color,
+      })),
+    [computedItems],
+  );
+
+  if ((!companyName || !orgId || !tickets) && !loading) return null;
```
Comment (fabio-silva) [inline src/components/SupportTicketOverview/SupportTicketOverview.tsx:88]: Missing brackets

File: src/components/SupportTicketOverview/SupportTicketOverview.tsx [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+  const loading = useSelector(getOrgTicketsPending);
+  const companyName = useSelector(getUserCompanyName);
+  const styles = useStyles(getStyles);
+
+  const computedItems = useMemo(() => {
+    const categories = tickets.reduce((acc: ICategory[], ticket) => {
+      const alreadyInAcc = acc.find((category: ICategory) => category.department === ticket.department);
+
+      if (alreadyInAcc) {
+        // eslint-disable-next-line no-magic-numbers
```
Comment (fabio-silva) [inline src/components/SupportTicketOverview/SupportTicketOverview.tsx:41]: Same as the other file for disabling the rule

File: src/components/SupportTicketOverview/SupportTicketOverview.tsx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+      <h4>{Messages.header}</h4>
+      <div className={styles.container}>
+        {loading ? (
+          <Skeleton height={230} />
+        ) : (
+          <div className={styles.containerPadding}>
+            <Legend values={computedItems} />
+            <Pie
+              values={pieValues}
+              size={200}
```
Comment (fabio-silva) [inline src/components/SupportTicketOverview/SupportTicketOverview.tsx:101]: Let's move these numbers to constants

File: src/components/SupportTicketOverview/SupportTicketOverview.tsx [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+      return [
+        ...acc,
+        {
+          department: ticket.department,
+          ammount: 1,
+          color: '#fff',
+          types: [ticket.taskType],
+        },
+      ];
+    }, []);
```
Comment (fabio-silva) [inline src/components/SupportTicketOverview/SupportTicketOverview.tsx:65]: Just to be sure: this `computedItems` is supposed to be the same across renders?

**Review by fabio-silva — CHANGES_REQUESTED** [review]
Besides all the other points, I realise we're mixing `className` with `styles`. That can be avoided, like this: https://github.com/percona-platform/grafana/blob/main/public/app/percona/integrated-alerting/components/Severity/Severity.styles.ts

`stylesFactory` is considered deprecated (it does memoization), but we can still return a style object based on props without adding separate styles.

File: src/components/SupportTicketOverview/SupportTicketOverview.tsx (hunk 8 above)
Comment (TomasBlazek3PG) [reply src/components/SupportTicketOverview/SupportTicketOverview.tsx:56]: This is not the dependency array of useMemo. Its the accumulator of reduce function.

Comment (fabio-silva) [reply src/components/SupportTicketOverview/SupportTicketOverview.tsx:60]: Oh, right, my bad!

File: src/store/orgs/org.actions.ts [hunk 9]
```diff
@@ -160,9 +160,10 @@ export const searchOrgsAction = createAsyncThunk<Organization[], void>(
   'ORGS:ORGANIZATIONS/SEARCH',
   async (_, { rejectWithValue }) => {
     try {
-      const { data } = await OrgAPI.searchOrgs();
+      /* const { data } = await OrgAPI.searchOrgs(); */
+      const data = await OrgAPI.searchOrgs();
 
-      return transformOrganizations(data.orgs);
+      return transformOrganizations(data.data.orgs);
```
Comment (fabio-silva) [inline src/store/orgs/org.actions.ts:166]: Can we remove this comment? Also, better do `result = await ...`, because `data.data` is a bit odd

**Review by fabio-silva — CHANGES_REQUESTED** [review]

**Review by fabio-silva — APPROVED** [review]

**Review by nicolalamacchia — APPROVED** [review]

**Review by fabio-silva — APPROVED** [review]

**Pull request merged**